information/specification.md

```diff
@@ -0,0 +1,706 @@
+
+ * The most obvious strategy for coin selection, which consists of trying to
+   get as close to the requested value as possible, will tend (over time) to
+   create a lot of _dust_: small unspent outputs.
+
+   Dust outputs are a problem, because even if the total value of dust in a
+   wallet is more than enough to cover a given target amount, if we attempt to
+   include that dust in a given transaction, we may run out of space (by
+   reaching the transaction size limit) before we can cover the target amount.
+
+ * The most obvious strategy for paying change, which consists of making a
+   single change output with the exact excess value, will tend (over time) to
+   reduce the size of the UTxO set. This is bad for two reasons:
+
+    1. Having a small UTxO set limits the number of payments that we can make
+       in parallel. Since a single UTxO entry can only be used to pay for a
+       single output, we need at least as many UTxO entries as there are
+       outputs.
+
+    2. The approach of coalescing all change into a single output is widely
+       considered to have negative privacy implications. The discussion of
+       privacy issues is beyond the scope of this article.
+
+### Goals
+
+There are several desirable properties that we would like a coin selection
+algorithm to have. These properties include:
+
+ * A coin selection algorithm should, over the course of time, aim to generate
+   and maintain a UTxO set with _useful_ outputs: that is, outputs that allow
+   us to process future payments with a _minimum_ number of inputs.
+
+ * A coin selection algorithm should employ strategies to limit the
+   amount of _dust_ that accumulates in the UTxO set.
+
+For more information on dust avoidance, see [Self Organisation in Coin
+Selection](#self-organization-in-coin-selection).
+
+# Definitions
+
+### Address
+
+An _address_ is a unique identifier that represents a destination for a payment.
+
+Addresses are owned (and generated) by individual wallets.
+
+In general, coin selection algorithms are agnostic to the choice of address
+format. Therefore, the details of individual address formats are not described
+within this document.
+
+### Coin Amount
+
+A _coin amount_ is a non-negative integer value that represents a number of
+[Lovelace](https://cardanodocs.com/cardano/monetary-policy/).
+
+One [Ada](https://cardanodocs.com/cardano/monetary-policy/) is _exactly_ equal
+to one million Lovelace.
+
+### UTxO Set
+
+A _UTxO set_ represents the unspent value associated with a wallet.
+
+Each member of the set is a pair of the form (**_i_**, **_o_**), where:
+
+  * **_i_** is a [transaction input](#transaction-input);
+  * **_o_** is a [transaction output](#transaction-output).
+
+A given [transaction input](#transaction-input) can only appear _once_ in a
+given UTxO set.
+
+### Transaction Input
+
+A _transaction input_ is a pair of values (**_h_**, **_n_**), where:
+
+  * **_h_** is the hash of an existing transaction **_t_**;
```

Comment:
Any reason why this is hash and not transaction id, which is more general?

You could add a definition for Transaction ID as a unique identifier that represents a transaction. Coin selection algorithms are agnostic to the format of txid, but in cardano mainnet, it's the blake2b-256 hash of the cbor-encoded transaction.

Reply:
Agree, it doesn't quite matter what `h` is here (a hash, an encrypted digest .. whatever) but what's important though is that `h` must be a **unique** transaction identifier.

Reply:
@rvl wrote:
> Any reason why this is hash and not transaction id, which is more general?

@KtorZ also wrote:
> it doesn't quite matter how h is calculated here (a hash, an encrypted digest .. whatever) but what's important though is that h must be a unique transaction identifier.

Thanks for this. I've fixed this in the following change: a76cc0b6201e3e52cef6c2239517558621d7502d

I've also added this to the [list of types that can be generalized](https://github.com/input-output-hk/cardano-coin-selection/issues/17) within `cardano-coin-selection`.